pkg/resources/mutation/cluster/cluster.go

```diff
@@ -0,0 +1,132 @@
+		return psa.AdmissionResponseAllowed(), nil
+	}
+	oldCluster, newCluster, err := objectsv3.ClusterOldAndNewFromRequest(&request.AdmissionRequest)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get old and new clusters from request: %w", err)
+	}
+	if newCluster.Name == "local" {
+		return psa.AdmissionResponseAllowed(), nil
+	}
+	if newCluster.Spec.RancherKubernetesEngineConfig == nil {
+		// do nothing and leave it to the validator to reject the request
+		return psa.AdmissionResponseAllowed(), nil
+	}
+	newTemplateName := newCluster.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName
+	oldTemplateName := oldCluster.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName
+
+	// If the template is set(or changed), update the cluster with the new template's content
+	if newTemplateName != "" {
+		err := m.setPSAConfig(newCluster)
+		if err != nil && !apierrors.IsNotFound(err) {
+			return nil, fmt.Errorf("failed to set PSAconfig: %w", err)
+		}
+	} else {
+		switch request.Operation {
+		case admissionv1.Create:
+			return psa.AdmissionResponseAllowed(), nil
+		case admissionv1.Update:
+			// It is a valid use case where user switches from using PSACT to putting a PluginConfig for PSA under kube-api.AdmissionConfiguration,
+			// but it is not a valid use case where the PluginConfig for PSA has the same content as the one in the previous-set PSACT,
+			// so we need to drop it in this case.
+			if oldTemplateName != "" {
+				newConfig, found := psa.GetPlugConfigFromCluster(newCluster)
+				if found {
+					// found means there is a Plugin Config for PSA under the kube-api.admission_configuration section
+					oldConfig, _ := psa.GetPlugConfigFromCluster(oldCluster)
+					if reflect.DeepEqual(newConfig, oldConfig) {
+						psa.DropPSAPlugConfigFromAdmissionConfig(newCluster)
+					}
+				}
+			}
+		}
+	}
+
+	response := &admissionv1.AdmissionResponse{}
+	if err := patch.CreatePatch(request.Object.Raw, newCluster, response); err != nil {
+		return response, fmt.Errorf("failed to create patch: %w", err)
+	}
+	response.Allowed = true
+	return response, nil
+}
+
+// setPSAConfig makes sure that the PodSecurity config under the admission_configuration section matches the
+// PodSecurityAdmissionConfigurationTemplate set in the cluster
+func (m *ManagementClusterMutator) setPSAConfig(cluster *apisv3.Cluster) error {
+	template, err := m.psact.Get(cluster.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName)
+	if err != nil {
+		return fmt.Errorf("failed to get PodSecurityAdmissionConfigurationTemplate: %w", err)
+	}
+	plugin, err := psa.GetPluginConfigFromTemplate(template)
+	if err != nil {
+		return fmt.Errorf("failed to get plugin config from template: %w", err)
+	}
+	admissionConfig := psa.GetAdmissionConfigFromCluster(cluster)
+	found := false
+	for i, item := range admissionConfig.Plugins {
+		if item.Name == "PodSecurity" {
+			admissionConfig.Plugins[i] = plugin
+			found = true
```

Comment:
maybe break from the loop after it's found?

Reply:
the `break` is added to save some iterations 😄